tf2_ros/include/tf2_ros/async_buffer_interface.h

```diff
@@ -39,11 +40,44 @@
 #include <functional>
 #include <future>
 #include <string>
+#include <utility>
 
 namespace tf2_ros
 {
 
-using TransformStampedFuture = std::shared_future<geometry_msgs::msg::TransformStamped>;
+class TransformStampedFuture : public std::shared_future<geometry_msgs::msg::TransformStamped>
+{
+  typedef std::shared_future<geometry_msgs::msg::TransformStamped> _Base_type;
+
+public:
+  /// Constructor
+  explicit TransformStampedFuture(_Base_type && future) noexcept
+  : _Base_type(std::move(future)) {}
+
+  /// Copy constructor
+  TransformStampedFuture(const TransformStampedFuture & ts_future) noexcept
+  : _Base_type(ts_future),
```

Comment:
Just curious: what does this type cast achieve for all the constructors in this class?

Reply:
I don't think it is a type cast; instead, I think it is calling the constructor of the base class with the argument.  @iuhilnehc-ynos can confirm, though.